### ISSUE TITLE
background is not completely black

### ISSUE THREAD

**Issue by maxstepin** [tomolt/libschrift#7]
I'm using SFT_CHAR_IMAGE flag to get a bitmap, and it all looks fine at first. I'm getting a nice white glyph on black background. But then I noticed the background is not always (0,0,0) black. There are plenty of (1,1,1) pixels to the right side of the glyph. Barely visible horizontal lines going from the right side of the glyph up to the right edge of the bitmap.

**Comment by tomolt** [comment]
Hi maxstepin.
This is a problem I have known of pretty much from the start. It's not really a bug, and more a limitation of the underlying algorithm. These rows not-quite-black pixels are the result of a combination of quantization errors and general numerical instability.
It is very likely that you can find the same behaviour with other font renderers that use a similiar algorithm such as stb_truetype.
Is this only an annoyance to you, or does this actually pose a potential problem?

**Comment by maxstepin** [comment]
Yeah, it's not really a problem. 
Thanks!

**Comment by maxstepin** [comment]
I tried switching to floating point here:
struct cell    { double area, cover; };
and invoke quantize() at the last possible moment in post_process()
and artifacts seems to disappear completely.

Just thought you might consider it.

I looked into stb_truetype and they also seems to be doing float->int conversion at the last possible moment.

**Comment by tomolt** [comment]
Hi maxstepin,

So I just tested this for myself. And you're right, the background is now completely black!

I would've thought the error caused by numerical instability would be much greater.
Actually, last time I measured it it was in fact bigger than the quantization error!
But since then, the algorithm has been changed dramatically, so I really should have updated my measurements.

I was a bit worried about the performance implications, but apparently this is pretty much exactly as fast as before. This is pretty shocking, considering we're now reading and writing 4x the amount of data to the buffer! Just in the last major version release, I got a 15% speedup overall by reducing the data size of the outline struct - I would have expected a comparable slowdown here. I guess `draw_lines` and `post_process` just aren't that memory-bound in the first place. Oh well, at least this turns out to be an easy fix now!

I'll probably put this in the next major version (v7.1).
Thanks!

SideNote: I also tested this with float instead of double, but the slowdown from double-to-float conversion makes this *much* slower, so it's currently not worth it.
